From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: etil2jz <blanchot.arthur@protonmail.ch>
Date: Fri, 22 Apr 2022 19:02:16 +0200
Subject: [PATCH] Fix missing breaking causes for armor stands

Armor stands should be destroyed by falling anvils,
cactus and lava. See https://bugs.mojang.com/browse/MC-203302.

diff --git a/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java b/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
index e5ef24d92de21c4c0e6a98e06985e52d47bfdce0..fca957f698579104ac16cd0ce6496fe330eefcc8 100644
--- a/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
+++ b/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
@@ -511,6 +511,17 @@ public class ArmorStand extends LivingEntity {
                 } else if (DamageSource.ON_FIRE.equals(source) && this.getHealth() > 0.5F) {
                     this.causeDamage(source, 4.0F);
                     return false;
+                // Mirai start - fix missing breaking causes for armor stands
+                } else if (DamageSource.LAVA.equals(source)) {
+                    this.causeDamage(source, 4.0F);
+                    return false;
+                } else if (DamageSource.CACTUS.equals(source)) {
+                    this.causeDamage(source, amount);
+                    return false;
+                } else if (DamageSource.ANVIL.equals(source)) {
+                    this.causeDamage(source, amount * 3.0F);
+                    return false;
+                // Mirai end
                 } else {
                     boolean flag = source.getDirectEntity() instanceof AbstractArrow;
                     boolean flag1 = flag && ((AbstractArrow) source.getDirectEntity()).getPierceLevel() > 0;
